src/models/AppState.js

```diff
@@ -39,6 +39,8 @@ function (Okta, Q, Factor, BrowserFeatures, Errors) {
   var securityImageUrlTpl = compile('{{baseUrl}}/login/getimage?username={{username}}');
 
   function getSecurityImage (baseUrl, username, deviceFingerprint) {
+    // Any "+" symbols in the username must be escaped, since it's considered a special character in the query string
+    username = username.replace(/\+/g, '%2B');
```

Comment:
might be easier to use `encodeURIComponent` so we cover all possible scenarios. Or is it really just `+` that can cause a problem with the username?

Reply:
+1 to using a full encode

Might as well cover our bases. If a bug creates users with `&` then we open ourselves up to `&username=foobar`, not that there's much risk in it besides breaking the image.

Reply:
Good catch. Just fixed it.

Reply:
@royalchan-okta Encode the full query string? How would a bug introduce a `&` into the param?